### ISSUE TITLE
Output warning if `tls` options are used when `tls.enabled` is `false`

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#11953]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Users have been confused why `tls` options have not been applied when neglecting to set `tls.enabled` to `true` in addition to setting the `tls` options on a component.

### Proposal

At the least we should output a warning, but we could also consider defaulting to `true` whenever `tls` options are specified, but `enabled` is unspecified (still allowing it to be explicitly specified to disable by setting to `false`).

### Version

vector 0.20.0 (x86_64-apple-darwin 2a706a3 2022-02-10)

**Comment by zamazan4ik** [comment]
@jszwedko could you please explain why do we need `tls.enabled` field? Because now I cannot understand the use case of this field. FMPOV, if any `tls.*` option is specified - TLS is enabled, otherwise - TLS is disabled.

Maybe we need just delete `tls.enabled` option?

**Comment by jszwedko** [comment]
I think letting users set `tls.enabled` to `false` even when other options are set is useful, but I'm open to defaulting `tls.enabled` to `true` whenever any other setting is present.

**Comment by zamazan4ik** [comment]
> I think letting users set tls.enabled to false even when other options are set is useful

What is the use-case for this behaviour? I see only one - temporarily disable tls but preserve TLS settings in the config file. But this scenario is perfectly covered by just commenting TLS settings.

**Comment by jszwedko** [comment]
Right, yeah, I was thinking temporarily disabling the options or using environment variables to toggle it on or off in different environments.

**Comment by zamazan4ik** [comment]
> Right, yeah, I was thinking temporarily disabling the options or using environment variables to toggle it on or off in different environments.

It was easy to guess :D

As  I said, this use-case is already covered with commenting redundant options in config file. If we look to other quite similar places in Vector configuration, we also can see some inconsistency in this question:
- https://vector.dev/docs/reference/configuration/sources/aws_s3/#auth - has no `enabled` option
- https://vector.dev/docs/reference/configuration/sources/kafka/#sasl - has `enabled` option

So from my perspective I suggest to discuss these inconsistencies internally in Vector dev team first - I am sure you will find the best approach :)

**Comment by zamazan4ik** [comment]
Okay, I have another one :)

https://vector.dev/docs/reference/configuration/sources/prometheus_scrape/#auth

**Comment by jszwedko** [comment]
Mmm, yes 😄 To be consistent, my inclination is that should have `enabled` as well, defaulting to `true` if any other option is present.

**Comment by zamazan4ik** [comment]
I propose to discuss the question with other Vector' dev team and decide, what should we do. Since the decision potentially will triiger bunch of refactorings in other sources/sinks.
